src/lib/plugin.js

```diff
@@ -24,78 +24,64 @@ const assertOptionType = (opts, field, type) => {
   }
 }
 
-module.exports = class PluginVirtual extends EventEmitter2 {
+const moduleName = (paymentChannelBackend) => {
+  const pluginName = paymentChannelBackend.pluginName || 'payment-channel'
+  return 'ilp-plugin-' + pluginName.toLowerCase()
+}
 
-  constructor (opts) {
+module.exports = class PluginPaymentChannel extends EventEmitter2 {
+  constructor (paymentChannelBackend, opts) {
     super()
+    const Backend = getBackend(opts._store)
 
-    assertOptionType(opts, 'currencyCode', 'string')
-    assertOptionType(opts, 'currencyScale', 'number')
-    assertOptionType(opts, 'maxBalance', 'string')
-    assertOptionType(opts, 'secret', 'string')
-    assertOptionType(opts, 'peerPublicKey', 'string')
-    assertOptionType(opts, '_store', 'object')
-    assertOptionType(opts, 'rpcUri', 'string')
-
-    this._secret = opts.secret
-    this._peerPublicKey = opts.peerPublicKey
-    this._publicKey = Token.publicKey(this._secret)
-    this._currencyCode = opts.currencyCode.toUpperCase()
-    this._currencyScale = opts.currencyScale
-
-    this._store = opts._store
-    this._maxBalance = opts.maxBalance
-    this._balance = new Balance({
-      maximum: this._maxBalance,
-      store: this._store
-    })
+    this._opts = opts
+    this._stateful = !!(opts._backend || opts._store)
+    this.debug = paymentChannelBackend
+      ? debug(moduleName(paymentChannelBackend))
+      : debug('ilp-plugin-virtual')
```

Comment:
This is for the case where there's no paymentChannelBackend. We haven't thought of the name for that kind of plugin, because it's simpler than plugin virtual (which does ECDH stuff)